packages/dd-trace/src/appsec/waf/waf_manager.js

```diff
@@ -51,20 +54,27 @@ class WAFManager {
     return wafContext
   }
 
-  update (newRules) {
-    try {
-      this.ddwaf.update(newRules)
+  setRulesVersion () {
+    if (this.ddwaf.diagnostics.ruleset_version) {
+      this.rulesVersion = this.ddwaf.diagnostics.ruleset_version
+    }
+  }
 
-      if (this.ddwaf.diagnostics.ruleset_version) {
-        this.rulesVersion = this.ddwaf.diagnostics.ruleset_version
-      }
+  setAsmDdFallbackConfig () {
+    if (!this.ddwaf.configPaths.some(cp => cp.includes('ASM_DD'))) {
+      this.updateConfig(WAFManager.defaultWafConfigPath, this.defaultRules)
+    }
+  }
 
-      Reporter.reportWafUpdate(this.ddwafVersion, this.rulesVersion, true)
-    } catch (error) {
-      Reporter.reportWafUpdate(this.ddwafVersion, 'unknown', false)
+  updateConfig (path, rules) {
+    const updateResult = this.ddwaf.createOrUpdateConfig(rules, path)
+    this.setRulesVersion()
+    return updateResult
+  }
 
-      throw error
-    }
+  removeConfig (path) {
+    this.ddwaf.removeConfig(path)
+    this.setRulesVersion()
   }
```

Comment:
removeConfig also returns a boolean, why not handle it just like updateConfig() ?

Reply:
`removeConfig` returns true if configuration was removed successfully, and false if it does not exist. There is no diagnostics to extract errors or warnings from, so there is nothing to report.